internal/app/db/storage/postgres/order.go

```diff
@@ -0,0 +1,165 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/AsakoKabe/gophermart/internal/app/db/models"
+)
+
+type OrderStorage struct {
+	db *sql.DB
+}
+
+func NewOrderStorage(db *sql.DB) *OrderStorage {
+	return &OrderStorage{db: db}
+}
+
+type sqlRow interface {
+	Scan(dest ...any) error
+}
+
+const insertOrder = "insert into orders (num, user_id) values ($1, $2)"
+const selectOrderByNum = "select id, num, status, accrual, user_id, trim('\"' from to_json(uploaded_at)::text) from orders where num = $1"
+const selectOrdersByUserID = "select id, num, status, accrual, user_id, trim('\"' from to_json(uploaded_at)::text) from orders where user_id = $1 order by uploaded_at"
+const selectOrdersWithStatuses = "select id, num, status, accrual, user_id, trim('\"' from to_json(uploaded_at)::text) from orders where status in (%s)"
```

Comment:
```suggestion
const selectOrdersWithStatuses = "select id, num, status, accrual, user_id, trim('\"' from to_json(uploaded_at)::text) from orders where status = ANY($1)"
```
И в аргументы можно пробросить слайс из статусов, он будет автоматически преобразован в нужную форму.